src/Components/CreateImageWizardV2/steps/Review/Footer.tsx

```diff
@@ -55,20 +57,33 @@ const ReviewWizardFooter = () => {
   const onSave = async () => {
     const requestBody = await getBlueprintPayload();
     setIsOpen(false);
-    !composeId &&
+    if (composeId) {
       requestBody &&
-      createBlueprint({ createBlueprintRequest: requestBody });
+        updateBlueprint({ id: composeId, createBlueprintRequest: requestBody });
+    } else {
+      requestBody && createBlueprint({ createBlueprintRequest: requestBody });
+    }
   };
 
   const onSaveAndBuild = async () => {
     const requestBody = await getBlueprintPayload();
     setIsOpen(false);
-    const blueprint =
-      !composeId &&
+    if (composeId) {
```

Comment:
This is getting a bit hard to understand, we have multiple conditionals in series and they have nested conditionals in the form of short circuit statements. Then there’s also the complicated async logic, including the call to `unwrap()` (it is not in this PRs scope but it would be nice to extract that into a function with a clear name to help future us know why we need to do that).

Could we try making this more composable to improve the readability? What if we had two components, `<CreateSubmitButton>` and `<EditSubmitButton>`. We could make the `submitButton` a prop for the ReviewFooter and pass in the appropriate one when creating the wizard. What do you think? Would you mind giving that a try?

Reply:
Sure thing 👍